gno.land/cmd/gnoland/start.go

```diff
@@ -372,3 +381,23 @@ func getTxEventStoreConfig(c *startCfg) (*eventstorecfg.Config, error) {
 
 	return cfg, nil
 }
+
+func initTelemetry(ctx context.Context) error {
+	var options []telemetry.Option
+
+	if os.Getenv("TELEM_METRICS_ENABLED") == "true" {
```

Comment:
which is the rationale in having metrics collection behind a feature flag? It would make the code heavier to read when everything is behind a feature flag such metrics.

Reply:
It's behind a feature flag currently because I think this will only be turned on for testing environments until we can verify that there is no significant performance impact under higher loads; just being cautious.

Reply:
I would recommend to move the feature flag inside the package itself rather than spam the high level packages which use it with if-condition. You can call still the `Collect()` method and when the collection is disabled then the `Collect()` method would not do any actual call to the OTEL SDK.

Reply:
I'm unsure which way to go.  I did it this way because the call to `telemetry.MetricsEnabled` will be inlined by the compiler and avoid the overhead of an unnecessary function call.  I guess the impact of the optimization depends on how many metrics we end up collecting.